pkg/crc/preflight/preflight_checks_nonlinux.go

```diff
@@ -0,0 +1,72 @@
+//go:build !linux
+// +build !linux
+
+package preflight
+
+import (
+	"runtime"
+	"strings"
+
+	"github.com/code-ready/crc/pkg/crc/daemonclient"
+	"github.com/code-ready/crc/pkg/crc/logging"
+	"github.com/shirou/gopsutil/process"
```

Comment:
Would be better to use `github.com/shirou/gopsutil/v3/process`

Reply:
@cfergeau I tired following and looks like it have some issue.
```golang
package main

import (
	"fmt"
	"github.com/shirou/gopsutil/v3/process"
	"log"
	"strings"
)

func main() {
		processes, err := process.Processes()
		if err != nil {
			log.Fatal(err)
		}
		for _ , process := range processes {
			name, err := process.Name()
			if err != nil {
				fmt.Println("Not able to get process Name")
				log.Fatal(err)
			}
			if strings.TrimSpace(name) == "crc" {
				fmt.Println("Got the process: ", name)
				cmdLine, err := process.CmdlineSlice()
				if err != nil {
					log.Fatal(err)
				}
				if cmdLine[1] == "daemon" {
					fmt.Println("Got the pid: ", process.Pid)
				}
			}
		}
}
```

Run it
```
$ go run proc.go
Not able to get process Name
2022/04/19 13:04:04 invalid argument
exit status 1
```

Reply:
> I tried the following and looks like it has some issue.

Your example works fine on my system, no errors when calling `process.Name()`, no idea what's going on with yours :-/

Reply:
I've been able to reproduce on macOS, and this is indeed behaves differently between v3 and !v3 :-/

Reply:
That's a bit of a mess:
https://github.com/shirou/gopsutil/issues/1288
It's even worse as Exe() does always for me, and is broken on !CGO_ENABLED builds, so we cannot fallback to this. Maybe the best for now is to ignore Name() errors and assume this was not `crc` when we have an error.

Reply:
> Maybe the best for now is to ignore Name() errors and assume this was not crc when we have an error.

@cfergeau as per https://github.com/shirou/gopsutil/issues/1288#issuecomment-1103214757 looks like https://github.com/shirou/gopsutil/pull/1229#issuecomment-1012408022 is what the issue. I am wondering why then `process.Processes()` have list of the process which is not owned by current user, should we first filter those process with current user and the check `process.Name()` ?

Reply:
Something like
```golang
package main

import (
	"fmt"
	"github.com/shirou/gopsutil/v3/process"
	"log"
	"os/user"
	"strings"
)

func main() {
		processes, err := process.Processes()
		if err != nil {
			log.Fatal(err)
		}
		for _ , process := range processes {
			usr, err := process.Username()
			if err != nil {
				log.Fatal(err)
			}
			currentUser, err := user.Current()
			if err != nil {
				log.Fatal(err)
			}
			if currentUser.Username != usr {
				continue
			}
			name, err := process.Name()
			if err != nil {
				log.Fatal(err)
			}
			if strings.TrimSpace(name) == "crc" {
				fmt.Println("Got the process: ", name)
				cmdLine, err := process.CmdlineSlice()
				if err != nil {
					log.Fatal(err)
				}
				if cmdLine[1] == "daemon" {
					fmt.Println("Got the pid: ", process.Pid)
				}
			}
		}
}
```

Reply:
@cfergeau ignored the error for `process.Name()`